client/app/services/group/currentGroup.service.js

```diff
@@ -14,7 +14,18 @@ export default class CurrentGroup {
     Object.assign(this, {
       value: {},
       User,
-      SessionUser
+      SessionUser,
+      map: {
```

Comment:
Although even watches should be deregistered:

```
$onInit() {
this.destroy = this.$scope.$watch(...);
}

$onDestroy() {
this.destroy();
}
```

I skipped doing this in at least one case, shame on me ;) You can see that watches are some kind of observer pattern, thy both come with registering and deregistering.